routing/async.go

```diff
@@ -0,0 +1,128 @@
+package routing
+
+import (
+	"context"
+	"sync"
+
+	"github.com/libp2p/go-libp2p-core/peer"
+	kbucket "github.com/libp2p/go-libp2p-kbucket"
+)
+
+// AsyncEvent is emitted for every notable event that happens during a DHT async lookup.
+// AsyncEvent supports JSON marshalling because all of its fields do, recursively.
+type AsyncEvent struct {
+	// ID is a unique identifier for the lookup instance
+	ID string
+	// Key is the Kademlia key used as a lookup target.
+	Key kbucket.ID
```

Comment:
you probably want to define your types here and not have core depend on kbucket

generally the dependency structure is
go-libp2p -> [subsystem] -> go-libp2p-core